auth/tests/unit/cognito-client-oidc.unit.test.ts

```diff
@@ -0,0 +1,64 @@
+import {Template} from "aws-cdk-lib/assertions";
+import {schema} from "yaml-cfn";
+import {describe, it, beforeAll} from "vitest";
+import {readFileSync} from "fs";
+import {load} from "js-yaml";
+
+let template: Template;
+
+describe("Set up the Cognito User Pool OIDC client", () => {
+
+    beforeAll(() => {
+        const yamlTemplate: any = load(readFileSync("template.yaml", "utf-8"), {
+            schema: schema,
+        });
+        template = Template.fromJSON(yamlTemplate)
+        console.log('template', template);
+
+    });
+
+    it("should refer to cognito user pool", () => {
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            UserPoolId: {
+                Ref: "CognitoUserPool",
+            },
+        });
+    });
+
+    it("has Cognito as IDP", () => {
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            SupportedIdentityProviders: ["COGNITO"],
+        });
+
+    });
+
+    it("has correct Allowed OAuth Flow set", () => {
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            AllowedOAuthFlows: ["code"],
+        });
+    });
+
+    it("has correct AllowedOAuthScopes", () => {
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            AllowedOAuthScopes: ["email", "openid"],
+        });
+    });
+
+    it("should not generate secrets", () => {
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            GenerateSecret: false,
+        });
+    });
+
+    it("has correct tokens validity", () => {
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            AccessTokenValidity: 3600, //one hr
+        });
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            IdTokenValidity: 3600, // one hr
+        });
+        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
+            RefreshTokenValidity: 31536000, //one year
+        });
```

Comment:
Are you able to do this all in one assertion?

```suggestion
        template.hasResourceProperties("AWS::Cognito::UserPoolClient", {
            AccessTokenValidity: 3600, //one hr
            IdTokenValidity: 3600, // one hr
            RefreshTokenValidity: 31536000, //one year
        });
```